File: Domain/Constants/OrderStatus.cs
namespace Domain.Constants
{
    public static class OrderStatus
    {
        /// <summary>
        /// Received - Đã nhận đồ
        /// Khách mang đồ đến. Nhân viên đã kiểm đồ, cân, ghi phiếu, tạo order.
        /// Chưa bắt đầu giặt. Đây là trạng thái khởi tạo (Initial State).
        /// </summary>
        public const string Received = "Received";

        /// <summary>
        /// Processing - Đang giặt
        /// Đồ đang trong quy trình: giặt → sấy → là ủi.
        /// </summary>
        public const string Processing = "Processing";

        /// <summary>
        /// Completed - Đã giặt xong
        /// Đồ đã hoàn tất: sạch – khô – đã gấp.
        /// Đang chờ khách đến lấy. Chưa thanh toán (nếu tiệm cho thanh toán khi nhận).
        /// </summary>
        public const string Completed = "Completed";

        /// <summary>
        /// Delivered - Đã giao cho khách
        /// Khách đến nhận đồ. Đã thanh toán (hoặc đã thu COD).
        /// Order được đóng lại.
        /// </summary>
        public const string Delivered = "Delivered";

        public static string[] GetAll()
        {
            return new[]
            {
                Received,
                Processing,
                Completed,
                Delivered
            };
        }

        public static bool IsValid(string status)
        {
            return GetAll().Contains(status);
        }
    }
}

File: laundry-fontend/src/constants/orderStatus.ts
/**
 * Order Status Constants - Đồng bộ với BE (Domain.Constants.OrderStatus)
 */
export const OrderStatus = {
	/** Received - Đã nhận đồ (Trạng thái khởi tạo) */
	Received: 'Received',
	/** Processing - Đang giặt */
	Processing: 'Processing',
	/** Completed - Đã giặt xong */
	Completed: 'Completed',
	/** Delivered - Đã giao cho khách */
	Delivered: 'Delivered',
} as const;

export type OrderStatusType = typeof OrderStatus[keyof typeof OrderStatus];

/**
 * Mapping status sang text tiếng Việt
 */
export const OrderStatusLabels: Record<OrderStatusType, string> = {
	[OrderStatus.Received]: 'Đã nhận đồ',
	[OrderStatus.Processing]: 'Đang giặt',
	[OrderStatus.Completed]: 'Đã giặt xong',
	[OrderStatus.Delivered]: 'Đã giao cho khách',
};

/**
 * Mapping status sang màu cho Badge
 */
export const OrderStatusColors: Record<OrderStatusType, string> = {
	[OrderStatus.Received]: 'blue',      // Xanh dương - Trạng thái khởi tạo
	[OrderStatus.Processing]: 'yellow',  // Vàng - Đang xử lý
	[OrderStatus.Completed]: 'green',    // Xanh lá - Hoàn thành
	[OrderStatus.Delivered]: 'gray',     // Xám - Đã đóng
};

/**
 * Lấy label tiếng Việt cho status
 */
export const getOrderStatusLabel = (status: string): string => {
	return OrderStatusLabels[status as OrderStatusType] || status;
};

/**
 * Lấy màu cho Badge dựa trên status
 */
export const getOrderStatusColor = (status: string): string => {
	return OrderStatusColors[status as OrderStatusType] || 'gray';
};

/**
 * Kiểm tra status có hợp lệ không
 */
export const isValidOrderStatus = (status: string): boolean => {
	return Object.values(OrderStatus).includes(status as OrderStatusType);
};

